client/asset/eth/multirpc.go

```diff
@@ -1109,35 +1118,46 @@ func (m *multiRPCClient) SubscribeFilterLogs(ctx context.Context, query ethereum
 	})
 }
 
-var compliantProviders = []string{
-	"linkpool.io",
-	"mewapi.io",
-	"flashbots.net",
-	"mycryptoapi.com",
-	"runonflux.io",
-	"infura.io",
-	"rivet.cloud",
-	"alchemy.com",
+const (
+	// Compliant providers
+	providerIPC         = "IPC"
```

Comment:
I figured we would list this IPC one and include it in `compliantProviders` so the special case `p.host == ipcHost` check in `checkProvidersCompliance` could be removed.

Also we can use `providerIPC` in place of `ipcHost` like in `connectProviders`.

The last thing I way eying for these consts was using `providerRivetCloud` instead of the string literal in `connectProviders`.